### PR TITLE
[Feature][connector-v2][Iceberg] sink plugin

### PR DESCRIPTION
**Pull request by immustard**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by immustard** [comment]
@hailin0 @EricJoy2048 PTAL

**Comment by liugddx** [comment]
Thanks for your contribution, Here are some comments.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SinkConfig.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package org.apache.seatunnel.connectors.seatunnel.iceberg.config;
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SinkConfig.java:19]: Please add apache license.

**Review by liugddx — COMMENTED** [review]
1. Please add document.
2. Please add e2e test case.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SinkConfig.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.apache.seatunnel.api.configuration.Options;
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+/**
+ *
+ *
+ * @author mustard
+ * @version 1.0
+ * Create by 2023-07-05
+ */
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SinkConfig.java]: Remove these.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DataConverter.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import org.apache.iceberg.data.Record;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+
+/**
+ *
+ *
+ * @author mustard
+ * @version 1.0
+ * Create by 2023-07-05
+ */
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DataConverter.java]: The same above

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DataConverter.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+public interface DataConverter {
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DataConverter.java:24]: The same above

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DefaultDataConverter.java [hunk 5]
```diff
@@ -0,0 +1,219 @@
+package org.apache.seatunnel.connectors.seatunnel.iceberg.data;
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DefaultDataConverter.java:18]: The same above

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DefaultDataConverter.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+import static org.apache.iceberg.types.Type.TypeID.*;
+
+/**
+ *
+ *
+ * @author mustard
+ * @version 1.0
+ * Create by 2023-07-05
+ */
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DefaultDataConverter.java]: The same above

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DefaultDataConverter.java [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+@RequiredArgsConstructor
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/data/DefaultDataConverter.java:51]: The same above

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/sink/IcebergSink.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ *
+ *
+ * @author mustard
+ * @version 1.0
+ * Create by 2023-07-05
+ */
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/sink/IcebergSink.java]: The same above

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/sink/IcebergSinkFactory.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+import org.apache.seatunnel.api.table.factory.TableSinkFactory;
+import org.apache.seatunnel.connectors.seatunnel.iceberg.config.SinkConfig;
+
+/**
+ *
+ *
+ * @author mustard
+ * @version 1.0
+ * Create by 2023-07-05
+ */
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/sink/IcebergSinkFactory.java]: The same above

**Comment by liugddx** [comment]
Run `mvn spotless:apply` before submitting the code.

**Comment by liugddx** [comment]
If you are contributing the connector code, please check that the following files are updated:

1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)

2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it

3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)

4. Update the [release-note](https://github.com/apache/seatunnel/blob/dev/release-note.md).

**Comment by immustard** [comment]
@liugddx PTAL

**Comment by liugddx** [comment]
> @liugddx PTAL

Fix CI error.

**Comment by immustard** [comment]
您好，您的邮件已经收到。

**Pull request closed without merging**